zkevm-circuits/src/tx_circuit.rs

```diff
@@ -0,0 +1,516 @@
+//! The transaction circuit implementation.
+
+// Naming notes:
+// - *_be: Big-Endian bytes
+// - *_le: Little-Endian bytes
+
+mod sign_verify;
+
+use crate::util::{random_linear_combine_word as rlc, Expr};
+use eth_types::{
+    geth_types::Transaction, Address, Field, ToBigEndian, ToLittleEndian, ToScalar, Word,
+};
+use ff::PrimeField;
+use group::GroupEncoding;
+use halo2_proofs::{
+    circuit::{AssignedCell, Layouter, Region, SimpleFloorPlanner},
+    plonk::{Advice, Circuit, Column, ConstraintSystem, Error},
+    poly::Rotation,
+};
+use itertools::Itertools;
+use lazy_static::lazy_static;
+use libsecp256k1;
+use log::error;
+use num::Integer;
+use num_bigint::BigUint;
+use rlp::RlpStream;
+use secp256k1::Secp256k1Affine;
+use sha3::{Digest, Keccak256};
+use sign_verify::{SignData, SignVerifyChip, SignVerifyConfig};
+pub use sign_verify::{POW_RAND_SIZE, VERIF_HEIGHT};
+use std::convert::TryInto;
+use std::marker::PhantomData;
+use subtle::CtOption;
+
+lazy_static! {
+    // Scalar
+    static ref SECP256K1_Q: BigUint = BigUint::from_slice(&[
```

Comment:
Can you provide the link of the source of this constant, plz?

Reply:
Done!  Thanks for the suggestion! https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/484/commits/e3a4615f7fdf2961909ae0eb66491236ed57a7d1